+dj/Connection.m

```diff
@@ -168,6 +169,9 @@ function reload(self)
                 if ~isempty(self.initQuery)
                     self.query(self.initQuery);
                 end
+
+                self.serverId = self.query('SELECT CONNECTION_ID() as id').id;
```

Comment:
good points - has been a while, will double check everything, though I do seem to recall the `.id` doing.

Reply:
On the `external-storage` branch we have some tests related to populate within the external storage tests. If you want to take a look, they are in `TestExternalFile`. https://github.com/datajoint/datajoint-matlab/blob/8848a0209560983a6615afd7200e914aeb8a9afb/tests/TestExternalFile.m#L79

Reply:
r.e. point 1:
My thinking on why this might fall into a recursive loop is consider the following definition for query in your PR:
```matlab
        function ret = query(self, queryStr, varargin)
            % dj.Connection/query - query(connection, queryStr, varargin) issue an
            % SQL query and return the result if any.
            % The same connection is re-used by all DataJoint objects.
            if ~self.isConnected
                self.connId=mym(-1, 'open', self.host, self.user, self.password, self.use_tls);
                if ~isempty(self.initQuery)
                    self.query(self.initQuery);
                end

                self.serverId = self.query('SELECT CONNECTION_ID() as id').id;
```
Based on the above, on first connection, it would go into if block, connect and then hit the line `self.serverId` and enter the query block once more likely infinitely or until hit a recursive limit, no? We might be fine if there is special handling on `get` of `isConnected` but don't recall off-hand.

Reply:
r.e. point2:
Hmmm, I haven't yet come across a use-case where this has worked in my dev. Could be related that I dev in R2016b to ensure backward-compatibility but generally, it is best to evaluate functions into a variable and then index into it (as wasteful as it is...).

Reply:
notes from 2020-10-16 dev:
r.e. recurse: adjust to direct mym calls
r.e. indexing: xref docs, if present >= 2016b leave idx; else var then idx